src/main/webapp/3Dscript.js

```diff
@@ -43,58 +61,63 @@ function initThreeJs() {
 
   fetchData();
 }
-
 /**
- * Creates the 2D terrain and points a light at it.
- * It also calls two other functions so it can wait on the fetch
- * request and significantly cut down rendering time with
- * geometry instancing.
-*/
-function addMap() {
+ * Plot the map centered around this run's first coordinate.
+ * @param {Array<Point>} pose
+ */
+function initMap(pose) {
   // Creates the 2D map.
   const loader = new THREE.TextureLoader();
-  let material = new THREE.MeshLambertMaterial({
+  const material = new THREE.MeshLambertMaterial({
     map: loader.load(
         'https://maps.googleapis.com/maps/api/staticmap' +
       '?format=png&center=' + pose[0].lat + ',' + pose[0].lng +
       '&zoom=18&size=500x500&key=' + apiKey),
   });
-  let geometry = new THREE.PlaneGeometry(50, 50);
+  const geometry = new THREE.PlaneGeometry(50, 50);
   const map = new THREE.Mesh(geometry, material);
   map.position.set(0, -4, 0);
   map.rotation.x = -Math.PI / 2;
   scene.add(map);
-
+}
+/**
+ * Creates the 2D terrain and points a light at it. It also calls two other
+ * functions so it can wait on the fetch request and significantly cut down
+ * rendering time with geometry instancing.
+ * @param {Array<Point>} poseToPlot
+ * @param {string} hexColor
+*/
+function addPoseData(poseToPlot, hexColor) {
   // Add the light to the scene.
   const light = new THREE.PointLight(0xffffff, 1, 0);
   light.position.set(0, 100, 0);
   scene.add(light);
-  plotTrajectory();
+  plotTrajectory(poseToPlot);
 
   // Adds pose data to the scene as one Geometry instance.
-  material = new THREE.MeshBasicMaterial({color: 'red'});
+  material = new THREE.MeshBasicMaterial({color: hexColor});
   geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
-  orientation = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation = new THREE.InstancedMesh(geometry, material, poseToPlot.length);
   orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
   scene.add(orientation);
   plotOrientation();
 }
 
 /**
  * Converts lla coordinates to local world coordinates.
- * @param {double} lat Latitude in degrees.
- * @param {double} lng Longitude in degrees.
- * @param {double} alt Altitude in meters.
- * @return {array} An array returning gps coordinates.
+ * @param {Array<Point>} pose The run to center the data around.
```

Comment:
I think it makes more sense to accept a single point as the origin, and let the callers use `pose[0]` as necessary.